src/timeunit.ts

```diff
@@ -3,61 +3,62 @@ import {isObject, isString} from 'vega-util';
-  export const UTCDAY: 'utcday' = 'utcday';
-  export const UTCDATE: 'utcdate' = 'utcdate';
-  export const UTCHOURS: 'utchours' = 'utchours';
-  export const UTCMINUTES: 'utcminutes' = 'utcminutes';
-  export const UTCSECONDS: 'utcseconds' = 'utcseconds';
-  export const UTCMILLISECONDS: 'utcmilliseconds' = 'utcmilliseconds';
-  export const UTCYEARMONTH: 'utcyearmonth' = 'utcyearmonth';
-  export const UTCYEARMONTHDATE: 'utcyearmonthdate' = 'utcyearmonthdate';
-  export const UTCYEARMONTHDATEHOURS: 'utcyearmonthdatehours' = 'utcyearmonthdatehours';
-  export const UTCYEARMONTHDATEHOURSMINUTES: 'utcyearmonthdatehoursminutes' = 'utcyearmonthdatehoursminutes';
-  export const UTCYEARMONTHDATEHOURSMINUTESSECONDS: 'utcyearmonthdatehoursminutesseconds' =
-    'utcyearmonthdatehoursminutesseconds';
+  MONTHDATE: 'monthdate',
+  MONTHDATEHOURS: 'monthdatehours',
+  HOURSMINUTES: 'hoursminutes',
+  HOURSMINUTESSECONDS: 'hoursminutesseconds',
+  MINUTESSECONDS: 'minutesseconds',
+  SECONDSMILLISECONDS: 'secondsmilliseconds',
+  QUARTER: 'quarter',
+  YEARQUARTER: 'yearquarter',
+  QUARTERMONTH: 'quartermonth',
+  YEARQUARTERMONTH: 'yearquartermonth',
+  UTCYEAR: 'utcyear',
+  UTCMONTH: 'utcmonth',
+  UTCDAY: 'utcday',
+  UTCDATE: 'utcdate',
+  UTCHOURS: 'utchours',
+  UTCMINUTES: 'utcminutes',
+  UTCSECONDS: 'utcseconds',
+  UTCMILLISECONDS: 'utcmilliseconds',
+  UTCYEARMONTH: 'utcyearmonth',
+  UTCYEARMONTHDATE: 'utcyearmonthdate',
+  UTCYEARMONTHDATEHOURS: 'utcyearmonthdatehours',
+  UTCYEARMONTHDATEHOURSMINUTES: 'utcyearmonthdatehoursminutes',
+  UTCYEARMONTHDATEHOURSMINUTESSECONDS: 'utcyearmonthdatehoursminutesseconds',
 
   // UTCMONTHDATE and UTCMONTHDATEHOURS always include 29 February since we use year 0th (which is a leap year);
-  export const UTCMONTHDATE: 'utcmonthdate' = 'utcmonthdate';
-  export const UTCMONTHDATEHOURS: 'utcmonthdatehours' = 'utcmonthdatehours';
-  export const UTCHOURSMINUTES: 'utchoursminutes' = 'utchoursminutes';
-  export const UTCHOURSMINUTESSECONDS: 'utchoursminutesseconds' = 'utchoursminutesseconds';
-  export const UTCMINUTESSECONDS: 'utcminutesseconds' = 'utcminutesseconds';
-  export const UTCSECONDSMILLISECONDS: 'utcsecondsmilliseconds' = 'utcsecondsmilliseconds';
-  export const UTCQUARTER: 'utcquarter' = 'utcquarter';
-  export const UTCYEARQUARTER: 'utcyearquarter' = 'utcyearquarter';
-  export const UTCQUARTERMONTH: 'utcquartermonth' = 'utcquartermonth';
-  export const UTCYEARQUARTERMONTH: 'utcyearquartermonth' = 'utcyearquartermonth';
-}
+  UTCMONTHDATE: 'utcmonthdate',
+  UTCMONTHDATEHOURS: 'utcmonthdatehours',
+  UTCHOURSMINUTES: 'utchoursminutes',
+  UTCHOURSMINUTESSECONDS: 'utchoursminutesseconds',
+  UTCMINUTESSECONDS: 'utcminutesseconds',
+  UTCSECONDSMILLISECONDS: 'utcsecondsmilliseconds',
+  UTCQUARTER: 'utcquarter',
+  UTCYEARQUARTER: 'utcyearquarter',
+  UTCQUARTERMONTH: 'utcquartermonth',
+  UTCYEARQUARTERMONTH: 'utcyearquartermonth'
+} as const;
+
+type ValueOf<T> = T[keyof T];
```

Comment:
Please extract this into utils.

Reply:
Do you have a place for storing these util types?
I looked into `utils` and it is like a mixed bags of typeguards, etc. but if that is a preferred place, sure.

Reply:
Yeah, let's put it in utils.